pkg/app/app.go

```diff
@@ -97,27 +127,29 @@ func RealMain(opts types.Options, d types.Deployer, tester types.Tester) (result
 	if opts.ShouldBuild() {
 		if err := writer.WrapStep("Build", d.Build); err != nil {
 			// we do not continue to up / test etc. if build fails
+			done <- true
 			return err
 		}
 	}
 
 	// ensure tearing down the cluster happens last, even if up or test fails.
-	defer func() {
-		if opts.ShouldDown() {
-			// TODO(bentheelder): instead of keeping the first error, consider
-			// a multi-error type
-			if err := writer.WrapStep("Down", d.Down); err != nil && result == nil {
-				result = err
-			}
+	if opts.ShouldDown() {
```

Comment:
Now that we are removing the defer, Down() should be last after Up() and Test()

Reply:
If I remove defer then we need to move the condition to the end of the function, otherwise it will try to down the cluster first and then will create the cluster. So keeping defer as it is.